click-spark.js

```diff
@@ -11,8 +10,8 @@ class ClickSpark extends HTMLElement {
     this.parentNode.addEventListener("click", this.clickEvent);
   }
 
-  detachedCallback() {
-    this.removeEventListener("click", this.clickEvent);
+  disconnectedCallback() {
+    this.parentNode.removeEventListener("click", this.clickEvent);
```

Comment:
🐛 Actually, I failed to account for that fact that by the time `disconnectedCallback` is invoked, the parent node has already changed (possibly none if the element was removed). So we'd need to cache the original parent node within `connectedCallback`:

```javascript
  connectedCallback() {
    this._parent = this.parentNode;
    this._parent.addEventListener("click", this.clickEvent);
  }

  disconnectedCallback() {
    this._parent.removeEventListener("click", this.clickEvent);
    delete this._parent;
  }
```